### DISCUSSION TITLE
Plugin failure due to upgrade from 2.8 to 3.0

### DISCUSSION THREAD

**Discussion by nabeelaccount** [Kong/kong#11913] [Help]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

kong 3.0

### Current Behavior

I have been using this script: https://github.com/bluekiri/kong-plugin-jwt-auth-token-validate/tree/master/kong/plugins/jwt-auth-token-validate

and now it stopped working after upgrading from kong version 2.8 to 3.0. Is there any changes to the script I need to do that would be the cause of this issue? I appreciate all the help I can get.

I understand that the base_plugin is DEPRECATED as shown https://docs.konghq.com/gateway/latest/plugin-development/custom-logic/#migrating-from-baseplugin-module but is there anything else I've missed?

There's no much logging produced. I get this back when I look at the logging:
time="2023-10-31T13:40:57Z" level=info msg="Retrying kong admin api client call after error" error="making HTTP request: Get \"https://localhost:8444/\": dial tcp [::1]:8444: connect: connection refused" logger=setup retries=11/60
When I try running it locally I get:
lua: handler.lua:2: module 'kong.plugins.base_plugin' not found

### Expected Behavior

Running pods per node - daemonset

### Steps To Reproduce

1. Create kong application using the following helm chart: https://artifacthub.io/packages/helm/kong/kong
2. Set helm version to: 2.19.0 (kong-ingress to version 2.8) and kong version to 3.0
3. Use the plugin from https://github.com/bluekiri/kong-plugin-jwt-auth-token-validate/tree/master/kong/plugins/jwt-auth-token-validate updating the BasePlugin module to the new format as shown below:
`  local JWTValidateHandler = {
    PRIORITY = 990, -- set the plugin priority, which determines plugin execution order
    VERSION = "0.1", -- version in X.Y.Z format. Check hybrid-mode compatibility requirements.
  }`

### Anything else?

logs error produced:
time="2023-10-31T13:40:59Z" level=info msg="Retrying kong admin api client call after error" error="making HTTP request: Get \"https://localhost:8444/\": dial tcp [::1]:8444: connect: connection refused" logger=setup retries=13/60

**Comment by tzssangglass** [comment]
As this is not a plugin we maintain, I have turned this issue into a discussion.

**Reply by tzssangglass** [reply]
it looks like the Kong admin API is disabled.

**Reply by nabeelaccount** [reply]
I've now enabled the kong admin by setting the chart value true on:
`admin:
  enabled: true`

However, I'm still seeing the same issue from the logs. Describing the pod produces the following message:

`  Normal   Pulled     94s (x3 over 116s)   kubelet            Container image "kong:3.0" already present on machine
  Normal   Created    94s (x3 over 116s)   kubelet            Created container proxy
  Warning  Unhealthy  94s (x7 over 111s)   kubelet            Readiness probe failed: Get "http://10.244.1.57:10254/readyz": dial tcp 10.244.1.57:10254: connect: connection refused`

**Comment by nabeelaccount** [comment]
What changes, if any, would I need to do to make the plugin work with kong version 3.x please?

**Comment by tzssangglass** [comment]
> Normal Pulled 94s (x3 over 116s) kubelet Container image "kong:3.0" already present on machine Normal Created 94s (x3 over 116s) kubelet Created container proxy Warning Unhealthy 94s (x7 over 111s) kubelet Readiness probe failed: Get "http://10.244.1.57:10254/readyz": dial tcp 10.244.1.57:10254: connect: connection refused

I did not find any issues related to the plugin from these error logs. I think you need to start kong in k8s correctly first.
